app/models/nested_facet.rb

```diff
@@ -3,7 +3,21 @@ class NestedFacet < OptionSelectFacet
 
   def initialize(facet, values)
     @sub_facet_key = facet["sub_facet_key"]
-    @is_main_facet = !!facet["sub_facet_key"]
+    @is_main_facet = !facet["sub_facet_key"].nil?
     super
   end
+
+  def facet_options
```

Comment:
Would be good to explain in the commit message what's happening in this commit. Quite a lot to follow otherwise.

Reply:
Seconded.